src/main/java/com/fundynamic/d2tm/game/math/BoundedVector2D.java

```diff
@@ -0,0 +1,33 @@
+package com.fundynamic.d2tm.game.math;
+
+
+public class BoundedVector2D<T extends Number> extends Vector2D {
```

Comment:
The `BoundedVector` is not a more specialized vector; it's a vector that can do _less_ than its ancestor. I think this causes the same problems as when you derive `Square` from `Rectangle`. It violates LSP and causes subtle bugs. And if you don't believe me, check what [Uncle Bob has to say about it](http://cafe.elharo.com/programming/a-square-is-not-a-rectangle/).

Imho, having a `Perimeter`-like class that constrains the vector and is applied separately, is _way_ better!